### PR TITLE
[network] error on receiving a noise handshake payload in FN network

### PR DESCRIPTION
**Pull request by mimoo**
In a full node network, or as we call it currently a ServerOnly handshake mode,
we do not send a payload during the handshake.
(In the validator network we do, for anti-replay mitigation.)
Thus, we should check on the responder/server side that the payload was empty,
and return an error otherwise.

### CODE REVIEW

File: network/src/noise/handshake.rs [hunk 1]
```diff
@@ -323,6 +324,12 @@ impl NoiseUpgrader {
 
             // store the timestamp
             anti_replay_timestamps.store_timestamp(their_public_key, client_timestamp);
+        } else if payload.len() != 0 {
+            // TODO: security logging
+            return Err(io::Error::new(
+                io::ErrorKind::InvalidData,
+                "noise: a handshake payload was received in the full-node network: {}",
+            ));
```
Comment (davidiw) [inline network/src/noise/handshake.rs:332]: I wonder if we already get this logging via the error same as above...

Comment (mimoo) [reply network/src/noise/handshake.rs:332]: yeah, and in any case it'll be cleaner to consolidate the logging on the caller side :o good point

Comment (phlip9) [reply network/src/noise/handshake.rs:332]: 1. probably remove the "{}" if you're not going to format anything : p

2. maybe call this "noise: received an unexpected handshake payload on a public-facing full-node network"?

**Comment by phlip9** [comment]
@bors-libra delegate+

land after fixing small nit : )

File: network/src/noise/handshake.rs (hunk 1 above)
Comment (mimoo) [reply network/src/noise/handshake.rs:332]: > probably remove the "{}" if you're not going to format anything : p

🤦🏼‍♂️

**Pull request closed without merging**